lib/features/children/generosity_challenge/pages/registration_redirect_to_generosity_screen.dart

```diff
@@ -38,6 +41,18 @@ class _RegistrationRedirectToGenerosityScreenState
     );
   }
 
+  void _goToChallenge(BuildContext context, UserExt user) {
+    AnalyticsHelper.logEvent(
+      eventName: AmplitudeEvents.goToChallengeFromRegistrationClicked,
+    );
+    final email = user.email;
+    getIt<SharedPreferences>().setString(
+      ChatScriptSaveKey.email.value,
+      email,
+    );
+    context.goNamed(FamilyPages.generosityChallenge.name);
+  }
+
```

Comment:
@TammiLion we are save the email before they start the challenge, because once they 'goNamed' the context and authCubit is wiped. Alternatively i'd have to pass the email through the rout and save it then.